Aikido.Zen.Core/Models/AgentContext.cs

```diff
@@ -124,6 +126,30 @@ public void AddRoute(Context context)
             _routes.AddOrUpdate(context.Route, route);
         }
 
+        /// <summary>
+        /// Adds a runtime package to the context.
+        /// This is used to track nuget packages once they are loaded by the application.
+        /// </summary>
+        /// <param name="packageName">The name of the package that was loaded</param>
+        /// <param name="packageVersion">The version of the package that was loaded</param>
+        public void AddRuntimePackage(string packageName, string packageVersion)
+        {
+            var identifier = $"{packageName.ToLowerInvariant()}@{packageVersion}";
+
+            _packages.AddOrUpdate(identifier, 
+                new Package
+                {
+                    Name = packageName,
+                    Version = packageVersion,
+                    RequiredAt = DateTimeHelper.UTCNowUnixMilliseconds()
```

Comment:
`AddOrUpdate` => it will use the latest `RequiredAt` time I guess?

See https://github.com/AikidoSec/firewall-node/blob/main/library/agent/Packages.ts#L15
See https://github.com/AikidoSec/firewall-node/blob/main/library/agent/Agent.ts#L320